server/middleware.go

```diff
@@ -0,0 +1,12 @@
+package server
+
+// Middlewares builds a middlware chain recursively.
+func Middlewares(next HandlerFunc, m ...func(HandlerFunc) HandlerFunc) HandlerFunc {
```

Comment:
I think this function should be a fixed type so it's easy to reference.
```go
type MiddlewareFunc func(HandlerFunc) HandlerFunc
```

I also think this whole file/function should be better documented both regarding details and regarding capitalization and ending dots. It's not super hard to understand but it goes better with the rest of the comments in my opinion!